### PR TITLE
decommission-node remote support

### PR DESCRIPTION
**Pull request by devidask27**
## ISSUE(S):

https://platform9.atlassian.net/browse/FT-350

## ISSUE TYPE

- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that may cause existing functionality to not work as expected)
- [] This change requires a documentation update

## IMPACTED FEATURES/COMPONENTS:

Impacted commands
decommission-node

#### Manual

**Remote decommission**
Screenshot 2022-03-01 at 4 44 44 PM
Screenshot 2022-03-01 at 4 45 10 PM

#### Reviewers

/cc @anupbarve @nikhilbandi

### CODE REVIEW

File: pkg/pmk/decomissionNode.go [hunk 1]
```diff
@@ -14,25 +12,17 @@ import (
...
-	output.Stdin = os.Stdin
-	err := output.Start()
-	output.Wait()
-	if err != nil {
-		fmt.Println(err.Error())
+func removePf9Instation(c client.Client) {
+	commands := map[string]string{
+		"Removing /etc/pf9 logs": "rm -rf /etc/pf9",
+		"Removing /opt/pf9 logs": "rm -rf /opt/pf9",
+		"Removing pf9 HOME dir":  "rm -rf $HOME/pf9",
```
Comment (anupbarve) [inline pkg/pmk/decomissionNode.go:19]: Possibly use the constants from here : https://github.com/platform9/pf9ctl/blob/master/pkg/util/constants.go#L92

File: pkg/pmk/decomissionNode.go [hunk 2]
```diff
@@ -50,11 +40,15 @@ func DecommissionNode(cfg *objects.Config, nc objects.NodeConfig, removePf9 bool
 	if err != nil {
 		zap.S().Debug("Failed to get keystone %s", err.Error())
 	}
+	ip, err := c.Executor.RunWithStdout("/bin/sh", "-c", "hostname -I")
+	ip = strings.TrimSpace(ip)
+	if err != nil {
+		zap.S().Debugf("unable to get host ip")
+	}
```
Comment (anupbarve) [inline pkg/pmk/decomissionNode.go:47]: Return in case of error ?

File: pkg/pmk/decomissionNode.go [hunk 3]
```diff
@@ -67,34 +61,43 @@ func DecommissionNode(cfg *objects.Config, nc objects.NodeConfig, removePf9 bool
...
+		"pkill kubelet",
+		"pkill etcd",
+		"pkill kube-proxy",
+		"pkill kube-apiserve",
+		"pkill kube-schedule",
+		"pkill kube-controll",
+		"rm -rf /opt/cni",
+		"rm -rf /opt/containerd",
+		"rm -rf /var/lib/containerd",
+		"rm -rf /var/opt/pf9",
```
Comment (anupbarve) [inline pkg/pmk/decomissionNode.go:95]: Put the list in constants.go, specifically the services and files.
You can construct the above commands i.e. pkill and rm dynamically in code using those constants.

File: pkg/pmk/decomissionNode.go [hunk 4]
```diff
@@ -67,34 +61,43 @@ func DecommissionNode(cfg *objects.Config, nc objects.NodeConfig, removePf9 bool
...
 
-		RunCommandWait("sudo yum erase -y pf9-comms")
-		RunCommandWait("sudo yum erase -y pf9-kube")
-		RunCommandWait("sudo yum erase -y pf9-hostagent")
-		RunCommandWait("sudo yum erase -y pf9-muster")
+		commands := []string{
+			"yum erase -y pf9-comms",
+			"yum erase -y pf9-kube",
+			"yum erase -y pf9-hostagent",
+			"yum erase -y pf9-muster",
```
Comment (anupbarve) [inline pkg/pmk/decomissionNode.go:74]: Put the list in constants.go, specifically the services.
You can construct the above command i.e. `yum erase` dynamically in code using those constants.

**Review by anupbarve — CHANGES_REQUESTED** [review]

File: pkg/pmk/decomissionNode.go (hunk 1 above)
Comment (devidask27) [reply pkg/pmk/decomissionNode.go]: Done

File: pkg/pmk/decomissionNode.go (hunk 2 above)
Comment (devidask27) [reply pkg/pmk/decomissionNode.go]: Added fatalf

File: pkg/pmk/decomissionNode.go (hunk 3 above)
Comment (devidask27) [reply pkg/pmk/decomissionNode.go]: Done

File: pkg/pmk/decomissionNode.go (hunk 4 above)
Comment (devidask27) [reply pkg/pmk/decomissionNode.go]: Done

**Comment by devidask27** [comment]
**Testing**
**Remote decommission centos**
cnt1
cnt2
cnt3
cnt4

**Remote decommission ubuntu**
ub1
ub2
ub3

**Local decommission ubuntu**
lc1
lc2
lc3

File: pkg/cmdexec/executor.go [hunk 5]
```diff
@@ -44,14 +44,16 @@ func (c LocalExecutor) RunCommandWait(command string) {
...
 
-func (r *RemoteExecutor) RunCommandWait(command string) {
+func (r *RemoteExecutor) RunCommandWait(command string) string {
 	o, err := r.RunWithStdout(command)
 	if err != nil {
 		zap.S().Debugf("Error :", err.Error())
 	}
-	fmt.Println(o)
+	return o
+	//fmt.Println(strings.TrimSpace(o))
```
Comment (anupbarve) [inline pkg/cmdexec/executor.go:56]: remove.

**Review by anupbarve — DISMISSED** [review]
lgtm

**Comment by anupbarve** [comment]
we should re-think on what is being displayed on console. It is mostly garbage from user's perspective. Maybe open another jira to clean it up.

**Review by anupbarve — APPROVED** [review]

**Pull request merged**